modules/react-components/src/grid/template-grid.tsx

```diff
@@ -0,0 +1,273 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ImageUtils, URLUtils } from "@wso2is/core/utils";
+import _ from "lodash";
+import React, { FunctionComponent, ReactElement, ReactNode, SyntheticEvent, useEffect, useState } from "react";
+import { Grid } from "semantic-ui-react";
+import { LinkButton } from "../button";
+import { TemplateCard, TemplateCardPropsInterface } from "../card";
+import { Heading } from "../typography";
+
+/**
+ * Proptypes for the template grid component.
+ */
+interface TemplateGridPropsInterface {
+    /**
+     * Empty placeholder
+     */
+    emptyPlaceholder?: ReactElement;
+    /**
+     * Heading for the grid.
+     */
+    heading?: ReactNode;
+    /**
+     * Callback to be fired on template selection.
+     */
+    onTemplateSelect: (e: SyntheticEvent, { id }: { id: string }) => void;
+    /**
+     * Enable/ Disable pagination.
+     */
+    paginate?: boolean;
+    /**
+     * Pagination limit.
+     */
+    paginationLimit?: number;
+    /**
+     * Grid pagination options.
+     */
+    paginationOptions?: TemplateGridPaginationOptionsInterface;
+    /**
+     * Sub heading for the grid.
+     */
+    subHeading?: ReactNode;
+    /**
+     * Title for the tags section.
+     */
+    tagsSectionTitle?: TemplateCardPropsInterface["tagsSectionTitle"];
+    /**
+     * List of templates.
+     */
+    templates: any;
```

Comment:
Let's type this properly. Please check other places as well.

Reply:
Fixed with commit `2ab4dc8`